src/ripple/protocol/STObject.h

```diff
@@ -401,7 +401,7 @@ class STObject
         if (!cf)
             throw std::runtime_error ("Wrong field type");
 
-        cf->setValue (value);
+        cf->setValue (std::move (value));
```

Comment:
Can we do this????  Do we know that `value` is an rvalue??  My intuition is telling me that `V` could be a non-const lvalue reference.  If that's so, then this `move` would leave the caller with surprisingly empty guts on their parameter.  But it's possible I've missed something.

My intuition tells me that if we want to do the `move` in appropriate situations we should be using a forwarding reference.  @HowardHinnant, could you weigh in on this?